app/src/main/java/com/gh4a/service/RepositoryService.java

```diff
@@ -0,0 +1,156 @@
+    }
+
+    public static Observable loadPullRequestCount(Activity activity, Repository repository, String state, boolean refresh) {
+        return Observable.create(emitter -> {
+            final String QUERY_FORMAT = "type:pr repo:%s/%s state:%s";
+            IssueService issueService = (IssueService)
+                    Gh4Application.get().getService(Gh4Application.ISSUE_SERVICE);
+            HashMap<String, String> filterData = new HashMap<>();
+            filterData.put("q", String.format(Locale.US, QUERY_FORMAT,
+                    repository.getOwner().getLogin(), repository.getName(), state));
+            try {
+                int count = issueService.getSearchIssueResultCount(filterData);
+                emitter.onNext(count);
+                emitter.onComplete();
+            } catch(IOException ioe) {
+                emitter.onError(ioe);
+            }
+        }).compose(RxTools.handle(activity, LOAD_PULL_REQUESTS_COUNT, refresh));
+    }
+
+    public static Observable updateStar(Gh4Application app, Activity activity, String repoOwner, String repoName, boolean isStarring) {
+        return Observable.create(e -> {
+            StarService starService = (StarService)
+                    Gh4Application.get().getService(Gh4Application.STAR_SERVICE);
+            RepositoryId repoId = new RepositoryId(repoOwner, repoName);
+            if (isStarring) {
+                starService.unstar(repoId);
+            } else {
+                starService.star(repoId);
+            }
+
+            e.onNext(!isStarring);
+            e.onComplete();
+        })
+        .compose(RxTools.handle(activity, UPDATE_STAR));
+    }
+
+    public static Observable updateWatch(Activity activity, String repoOwner, String repoName, boolean isWatching) {
+        return Observable.create(e -> {
+            WatcherService watcherService = (WatcherService)
+                    Gh4Application.get().getService(Gh4Application.WATCHER_SERVICE);
+            RepositoryId repoId = new RepositoryId(repoOwner, repoName);
+            if (isWatching) {
+                watcherService.unwatch(repoId);
+            } else {
+                watcherService.watch(repoId);
+            }
+
+            e.onNext(!isWatching);
+            e.onComplete();
+        })
+        .compose(RxTools.handle(activity, UPDATE_WATCH));
+    }
+
+    public static Observable isStarring(Activity activity, String repoOwner, String repoName, boolean refresh) {
+        return Observable.create(e -> {
+            StarService starService = (StarService)
+                    Gh4Application.get().getService(Gh4Application.STAR_SERVICE);
+            try {
+                boolean isStarring = starService.isStarring(new RepositoryId(repoOwner, repoName));
+                e.onNext(isStarring);
+                e.onComplete();
+            } catch(Exception ex) {
+                e.onError(ex);
+            }
+        })
```

Comment:
The basic pattern seems pretty repetitive here:

```
public static Observable<...> doSomething() {
    return Observable.create(emitter -> {
        try {
            result = doSomething();
            emitter.onNext(result);
            emitter.onComplete();
        } catch (Exception e) {
            emitter.onError(e);
        }
    })
    .compose(RxTools.handle(activity, id, refresh));
}
```

This feels like there must be a more elegant way to write this, with a base method of some sort.

(Also note how I clarified variable names ;-) )

Reply:
Effectively ... what about this 

```java
    public static Observable loadRepository(Activity activity, String repoOwner, String repoName,
            boolean refresh) {
        return RxTools.runCallable(
            () -> ((org.eclipse.egit.github.core.service.RepositoryService)
                Gh4Application.get().getService(Gh4Application.REPO_SERVICE))
                .getRepository(repoOwner, repoName),
            activity,
            LOAD_REPOSITORY,
            refresh
        );
    }
```
+ `RxTools` method 

```java
    public static Observable runCallable(Callable call, Activity activity, int loaderId, boolean refresh) {
        return Observable.fromCallable(call)
                .compose(RxTools.handle(activity, loaderId, refresh));
    }
```